spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -165,11 +180,14 @@ public void destroy() throws Exception {
 			Assert.state(binderConfiguration != null, "Unknown binder configuration: " + configurationName);
 			BinderType binderType = this.binderTypeRegistry.get(binderConfiguration.getBinderType());
 			Assert.notNull(binderType, "Binder type " + binderConfiguration.getBinderType() + " is not defined");
-			Map<Object, Object> binderProperties = binderConfiguration.getProperties();
+			
+			Map<String, String> binderProperties = new HashMap<>();
+			this.flatten("", binderConfiguration.getProperties(), binderProperties);
```

Comment:
Usual convention is to put called methods below callees. (and drop the `this.` on method calls).